### PR TITLE
MM-47283: Migrate /enterprise/system_console/channel_moderation e2e tests to TypeScript

### PR DESCRIPTION
**Pull request by adithyaakrishna**
#### Summary

This PR Migrate /enterprise/system_console/channel_moderation e2e tests to TypeScript

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/21332
Fixes https://mattermost.atlassian.net/browse/MM-47283

#### Release Note
```release-note
None
```

### CODE REVIEW

**Comment by AshishDhama** [comment]
@adithyaakrishna Could you, please fix the lint and type check issues.

File: e2e/cypress/tests/integration/enterprise/system_console/channel_moderation/channel_mentions_spec.ts [hunk 1]
```diff
@@ -24,10 +26,11 @@ import {
...
 
 describe('MM-23102 - Channel Moderation - Channel Mentions', () => {
-    let regularUser;
-    let guestUser;
-    let testTeam;
-    let testChannel;
+    let regularUser: UserProfile;
+    let guestUser: UserProfile;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let testTeam: any;
```
Comment (yasserfaraazkhan) [inline e2e/cypress/tests/integration/enterprise/system_console/channel_moderation/channel_mentions_spec.ts:32]: Hi @adithyaakrishna ,
 can we use Cypress.team and remove line 31

cypress.Chanel on line 33

**Review by yasserfaraazkhan — CHANGES_REQUESTED** [review]
everything looks good, but please cross the the types in few places

File: tsconfig.json [hunk 2]
```diff
@@ -21,6 +21,7 @@
     "experimentalDecorators": true,
     "jsx": "react",
     "baseUrl": ".",
+    "types": ["cypress"],
```
Comment (M-ZubairAhmed) [inline tsconfig.json:24]: can you please revert this change. This is not in scope of this pr also it can other side effects. https://www.typescriptlang.org/tsconfig#types
PS `e2e` have tsconfig too

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Review by furqanmlk — COMMENTED** [review]
@adithyaakrishna 
There are few errors.
Can you run ```e2e run check-types```  from /e2e/cypress directory

**Comment by adithyaakrishna** [comment]
Hey @M-ZubairAhmed 
Okay sure, I'll address the changes and send a commit regarding that :) 

@furqanmlk Yes, I will update and fix the errors with the new commit as well

File: e2e/cypress/tests/integration/enterprise/system_console/channel_moderation/channel_mentions_spec.ts (hunk 1 above)
Comment (adithyaakrishna) [reply e2e/cypress/tests/integration/enterprise/system_console/channel_moderation/channel_mentions_spec.ts]: Done 💯

File: tsconfig.json (hunk 2 above)
Comment (adithyaakrishna) [reply tsconfig.json]: Done 💯

**Comment by adithyaakrishna** [comment]
@AshishDhama Fixed the linting changes as well :)

**Review by M-ZubairAhmed — COMMENTED** [review]
Looks good. i just reviewed because it had changes to `tsconfig` which effects the webapp too. i will hand over to @furqanmlk and @yasserfaraazkhan for rest of the review

**Review by M-ZubairAhmed — APPROVED** [review]

File: e2e/cypress/tests/integration/enterprise/system_console/channel_moderation/helpers.ts [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
+import {UserProfile} from '@mattermost/types/users';
+
 import * as TIMEOUTS from '../../../../fixtures/timeouts';
 import {getAdminAccount} from '../../../../support/env';
 
 import {checkBoxes} from './constants';
 
 // # Visits the channel configuration for a channel with channelName
-export const visitChannelConfigPage = (channel) => {
+export const visitChannelConfigPage = (channel: { id?: string; name: string; create_at?: number; update_at?: number; delete_at?: number; team_id?: string; type?: ChannelType; display_name?: string; header?: string; purpose?: string; last_post_at?: number; last_root_post_at?: number; creator_id?: string; scheme_id?: string; teammate_id?: string; status?: string; group_constrained?: boolean; shared?: boolean; props?: Record<string, unknown>; policy_id?: string }) => {
```
Comment (yasserfaraazkhan) [inline e2e/cypress/tests/integration/enterprise/system_console/channel_moderation/helpers.ts:16]: @adithyaakrishna , 
just wondering if Cypress.Channel would work.

**Review by yasserfaraazkhan — CHANGES_REQUESTED** [review]
Please do check if line 16 and 72 of this file e2e/cypress/tests/integration/enterprise/system_console/channel_moderation/helpers.ts

can be altered and removing the _eslint-disabled-next-line_

File: e2e/cypress/tests/integration/enterprise/system_console/channel_moderation/helpers.ts (hunk 3 above)
Comment (adithyaakrishna) [reply e2e/cypress/tests/integration/enterprise/system_console/channel_moderation/helpers.ts:16]: @yasserfaraazkhan I guess it will, I will send a commit up to update it soon

**Comment by lieut-data** [comment]
Heads up that as part of our efforts to move to a monorepo, we're closing out a number of older pull requests like this one to help streamline the effort.

If you'd like to preserve these changes -- even if you're not the original author! -- feel free to resubmit the pull request against the monorepo once it's ready. You can subscribe to [mattermost-server-issue-22420](https://www.github.com/mattermost/mattermost-server/issues/22420) for status updates on this effort.
